### PR TITLE
Add oath2 login for web

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ all:
 	go test -v ./...
 	mkdir -p bin/
 	go build  -o bin/getcreds  -ldflags "-X main.Version=${VERSION}" cmd/getcreds/main.go
-	go build  -o bin/keymaster -ldflags "-X main.Version=${VERSION}" cmd/ssh_usercert_gen/main.go cmd/ssh_usercert_gen/templateData.go
+	go build  -o bin/keymaster -ldflags "-X main.Version=${VERSION}" cmd/ssh_usercert_gen/main.go cmd/ssh_usercert_gen/auth_oauth2.go cmd/ssh_usercert_gen/templateData.go
```
Comment (rgooch) [inline Makefile]: Why are you doing it this way? The convention is to have Go compile all the (non-test) code in a directory.

File: cmd/ssh_usercert_gen/auth_oauth2.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"golang.org/x/net/context"
+	//"golang.org/x/oauth2"
+	"io/ioutil"
+	"log"
```
Comment (rgooch) [inline cmd/ssh_usercert_gen/auth_oauth2.go]: Please consider using the logging support from the Dominator codebase and pass in the logger.

File: cmd/ssh_usercert_gen/auth_oauth2.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+		return
+	}
+
+	if r.URL.Query().Get("state") != pending.state {
+		http.Error(w, "state did not match", http.StatusBadRequest)
+		return
+	}
+	//if Debug {
+	//log.Printf("req : %+v", r)
+	//}
```
Comment (rgooch) [inline cmd/ssh_usercert_gen/auth_oauth2.go]: The Dominator codebase has a logger with Debug*() methods.

File: cmd/ssh_usercert_gen/auth_oauth2.go (hunk 2 above)
Comment (cviecco) [inline cmd/ssh_usercert_gen/auth_oauth2.go]: Fixed in latest push

File: cmd/ssh_usercert_gen/auth_oauth2.go (hunk 3 above)
Comment (cviecco) [inline cmd/ssh_usercert_gen/auth_oauth2.go]: We should adress this a a single PR that moves all loging there. Not piecemeal.

Comment (rgooch) [inline cmd/ssh_usercert_gen/auth_oauth2.go]: Yes.

**Pull request merged**